src/AdoPat/PatClient.cs

```diff
@@ -0,0 +1,143 @@
+    public class PatClient
+    {
+        private const int PageSize = 100; // Using the maximum allowable page size allows us to reduce HTTP calls.
+
+        // The AllOrgs field controls whether a PAT token create request is for all of a user's accessible organizations.
+        // The default of false means that the token is for a specific organization. The Azure DevOps team have
+        // recommended we always set this to false.
+        private const bool AllOrgs = false;
+
+        private ITokensHttpClientWrapper client;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PatClient"/> class.
+        /// </summary>
+        /// <param name="client">Any class which implements the <see cref="ITokensHttpClientWrapper"/> interface.</param>
+        public PatClient(ITokensHttpClientWrapper client)
+        {
+            this.client = client;
+        }
+
+        /// <summary>
+        /// Creates a new PAT.
+        /// </summary>
+        /// <param name="patTokenCreateRequest">
+        /// XXX: Not documented by Visual Studio client libraries. Should we depend on this or
+        /// should this method use the corresponding primitives as arguments? Not yet clear.
+        /// </param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>A <see cref="Task"/> returning a <see cref="PatTokenResult"/>.</returns>
+        public async Task<PatToken> CreatePatAsync(
+            PatTokenCreateRequest patTokenCreateRequest,
+            CancellationToken cancellationToken = default)
+        {
+            var patTokenResult = await this.client.CreatePatAsync(
+                patTokenCreateRequest,
+                cancellationToken: cancellationToken)
+            .ConfigureAwait(false);
+
+            if (patTokenResult.PatToken == null)
+            {
+                throw new PatClientException($"Failed to create PAT: {patTokenResult.PatTokenError}");
+            }
+
+            return patTokenResult.PatToken;
+        }
+
+        /// <summary>
+        /// Gets all active PATs.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>A mapping of authorization ID to active PATs.</returns>
+        public async Task<IDictionary<Guid, PatToken>> GetActivePatsAsync(CancellationToken cancellationToken = default)
+        {
+            // Initialize a PagedPatTokens so that we can use the continuation token
+            // in the scope of the conditional of the do while loop below. Azure
+            // DevOps will return the empty string when there are no more pages.
+            var pagedPatTokens = new PagedPatTokens(continuationToken: string.Empty, patTokens: new List<PatToken>());
+            var pats = new Dictionary<Guid, PatToken>();
+            do
+            {
+                pagedPatTokens = await this.client.ListPatsAsync(

```

Comment:
Did we discuss the null check here with Kyle? If not, we can move forward for now since we will not be shipping this until the PatManager is ready!

Reply:
@kyle-rader never did weigh in on that, but you, me, and @Haard30 reached something of a consensus I think. Let's roll with that for now with the knowledge that we can question it again before we ship the `PatManager` class.